src/pytorch_lightning/strategies/sharded.py

```diff
@@ -98,12 +102,13 @@ def _setup_model_and_optimizers(self, model: Module, optimizers: List[Optimizer]
         return model, optimizers
 
     def _wrap_optimizers(self, optimizers: List[Optimizer]) -> List["OSS"]:
-        if self.model is not None and self.model.trainer.state.fn != TrainerFn.FITTING:
+        assert self.lightning_module is not None
+        if self.model is not None and self.lightning_module.trainer.state.fn != TrainerFn.FITTING:
             return optimizers
 
         return self._reinit_optimizers_with_oss(optimizers)
 
-    def _reinit_optimizers_with_oss(self, optimizers: List[Union[Optimizer, LightningOptimizer]]) -> List["OSS"]:
+    def _reinit_optimizers_with_oss(self, optimizers: List[Optimizer]) -> List["OSS"]:
```

Comment:
same here

Reply:
yeah, it seems to be better typed as `List[Union[Optimizer, LightningOptimizer]]` just like before, but it gets error. i'm not sure to fix it or just keep this pr's changes
```
src/pytorch_lightning/strategies/sharded.py:109: error: Argument 1 to "_reinit_optimizers_with_oss" of "DDPShardedStrategy" has incompatible type "List[Optimizer]"; expected "List[Union[Optimizer, LightningOptimizer]]"  [arg-type]
src/pytorch_lightning/strategies/sharded.py:109: note: "List" is invariant -- see https://mypy.readthedocs.io/en/stable/common_issues.html#variance
src/pytorch_lightning/strategies/sharded.py:109: note: Consider using "Sequence" instead, which is covariant
```

Reply:
Hi, @rohitgr7, @lijm1358, are you fine with having it typed like this? Or do you have any other suggestion? It seems this conversation is the only blocker to have this PR merged

Reply:
Thank you, @otaj, I think it's fine with `List[Otpimizer]` as the conversation above (https://github.com/Lightning-AI/lightning/pull/14184#discussion_r949979509)

Reply:
looks like we don't need those `LightningOptimizer`check since we wrap the original Optimizer and all these wrapped optimizers is handled using `LightningOptimizer`. Let me push a commit and see if any tests fail.

Reply:
will do in a separate PR, let's merge this.